content/courses/native-onchain-development/cross-program-invocations.md

```diff
@@ -246,154 +253,156 @@ runtime to verify that the PDA belongs to the calling program is for the calling
+addresses. This feature is _extremely_ useful in many Web3 applications, such as
+DeFi and NFTs. Without CPIs, PDAs would be far less useful since programs
+wouldn't be able to sign transactions involving them—effectively turning them
+into black holes where assets sent to a PDA couldn't be retrieved without CPIs!
 
 ## Lab
 
-Now let's get some hands on experience with CPIs by making some additions to the
-Movie Review program again. If you're dropping into this lesson without having
-gone through prior lessons, the Movie Review program allows users to submit
-movie reviews and have them stored in PDA accounts.
+Now let's get some hands-on experience with CPIs by making some additions to the
+Movie Review program. If you're dropping into this lesson without going through
+the prior ones, the Movie Review program allows users to submit movie reviews,
+which are stored in PDA accounts.
 
-Last lesson, we added the ability to leave comments on other movie reviews using
-PDAs. In this lesson, we’re going to work on having the program mint tokens to
-the reviewer or commenter anytime a review or comment is submitted.
+In the
+[last lesson program derived addresses](/content/courses/native-onchain-development/program-derived-addresses.md),
+we added the ability to leave comments on movie reviews using PDAs. In this
+lesson, we'll work on having the program mint tokens to reviewers or commenters
+whenever a review or comment is submitted.
 
-To implement this, we'll have to invoke the SPL Token Program's `MintTo`
-instruction using a CPI. If you need a refresher on tokens, token mints, and
-minting new tokens, have a look at the
-[Token Program lesson](/content/courses/tokens/token-program) before moving
-forward with this lab.
+To implement this, we'll invoke the SPL Token Program's `MintTo` instruction
+using a CPI. If you need a refresher on tokens, token mints, and minting new
+tokens, check out the
+[Token Program lesson](/content/courses/tokens-and-nfts/token-program.md) before
+moving forward with this lab.
 
-#### 1. Get starter code and add dependencies
+### 1. Get starter code and add dependencies
 
-To get started, we will be using the final state of the Movie Review program
-from the previous PDA lesson. So, if you just completed that lesson then you’re
-all set and ready to go. If you are just jumping in here, no worries, you can
-[download the starter code here](https://github.com/Unboxed-Software/solana-movie-program/tree/solution-add-comments).
-We'll be using the `solution-add-comments` branch as our starting point.
+To get started, we'll be using the final state of the Movie Review program from
+the previous PDA lesson. If you just completed that lesson, you're all set and
+ready to go. If you're jumping in at this point, no worries! You can download
+the
+[starter code from the `solution-add-comments` branch](https://github.com/Unboxed-Software/solana-movie-program/tree/solution-add-comments).
```

Comment:
use https://github.com/solana-developers/movie-program/tree/solution-add-comments - check it has your updates